packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -358,11 +273,12 @@ const DataGrid = injectIntl((props) => {
       && !!(element.offsetWidth || element.offsetHeight || element.getClientRects().length)
       && window.getComputedStyle(element).visibility !== 'hidden'
       && element.closest('[inert]') === null
-      && (element.id === id || !grid.current.contains(element)),
+      && (element.id === id || !tableRef.current.contains(element)),
```

Comment:
We should not need to leverage the tableRef in this scenario.  We can check if the gridContainer contains the element.